apps/api/v2/src/modules/auth/strategies/api-auth/api-auth.strategy.spec.ts

```diff
@@ -0,0 +1,208 @@
+import { ApiKeyRepository } from "@/modules/api-key/api-key-repository";
+import { DeploymentsRepository } from "@/modules/deployments/deployments.repository";
+import { DeploymentsService } from "@/modules/deployments/deployments.service";
+import { JwtService } from "@/modules/jwt/jwt.service";
+import { OAuthClientRepository } from "@/modules/oauth-clients/oauth-client.repository";
+import { OAuthFlowService } from "@/modules/oauth-clients/services/oauth-flow.service";
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { PrismaWriteService } from "@/modules/prisma/prisma-write.service";
+import { RedisService } from "@/modules/redis/redis.service";
+import { TokensRepository } from "@/modules/tokens/tokens.repository";
+import { UsersRepository } from "@/modules/users/users.repository";
+import { ExecutionContext, HttpException } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { JwtService as NestJwtService } from "@nestjs/jwt";
+import { Test, TestingModule } from "@nestjs/testing";
+import { PlatformOAuthClient, Team, User } from "@prisma/client";
+import { ApiKeysRepositoryFixture } from "test/fixtures/repository/api-keys.repository.fixture";
+import { OAuthClientRepositoryFixture } from "test/fixtures/repository/oauth-client.repository.fixture";
+import { TeamRepositoryFixture } from "test/fixtures/repository/team.repository.fixture";
+import { TokensRepositoryFixture } from "test/fixtures/repository/tokens.repository.fixture";
+import { UserRepositoryFixture } from "test/fixtures/repository/users.repository.fixture";
+
+import { ApiAuthStrategy } from "./api-auth.strategy";
+
+describe("ApiAuthStrategy", () => {
+  let strategy: ApiAuthStrategy;
+  let userRepositoryFixture: UserRepositoryFixture;
+  let tokensRepositoryFixture: TokensRepositoryFixture;
+  let teamRepositoryFixture: TeamRepositoryFixture;
+  let organization: Team;
+  let oAuthClient: PlatformOAuthClient;
+  let apiKeysRepositoryFixture: ApiKeysRepositoryFixture;
+  let oAuthClientRepositoryFixture: OAuthClientRepositoryFixture;
+  const validApiKeyEmail = "api-key-user-email@example.com";
+  const validAccessTokenEmail = "access-token-user-email@example.com";
+  let validApiKeyUser: User;
+  let validAccessTokenUser: User;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        ApiAuthStrategy,
+        {
```

Comment:
NIT: I had a case like this in private API and extracted this to a ConfigServiceMock util something like. 

```ts
class ConfigServiceMock {
  private readonly config = {
    "db.readUrl": "postgresql://postgres:@localhost:5450/calendso",
    "db.writeUrl": "postgresql://postgres:@localhost:5450/calendso",
    "db.redisUrl": "redis://localhost:6379",
    "api.keyPrefix": "cal_test_",
    "e2e": true,
    "api.licenseKey": "license-key",
    "api.licenseKeyUrl": "license-key-url.cal.com",
    "env": "development"
  };

  get(key: string): any {
    return this.config[key] ?? null;
  }
}
```


Then the testing module would become 

```ts
   {
        provide: ConfigService,
        useClass: ConfigServiceMock,
      }
        ```

Reply:
the main reason we need to do this here is that we dont create a Nest App like we do in e2e specs, meaning that the env variables are not loading in the config service which is normally done in main.ts

Reply:
found a better way to write it